feat: Org-level Codacy/Slack integration TAROT-2301 (#1857)
codacy/docs: docs/organizations/integrations/slack-integration.md

# Organization Slack integration for Security issues

> This integration sends notifications to a Slack channel of your choice whenever Codacy detects new critical Security issues in any repository within your organization.

Reviewer: Using "This integration" kind of forces the reader to look at the title again to confirm the integration. (I did it! 😉)
Maybe we can use a "complete" sentence, as we do on other similar introductions (check [this one](https://tarot-2301-doc-document-slack-integ--docs-codacy.netlify.app/organizations/configuring-default-git-provider-integration-settings/), or [this one](https://tarot-2301-doc-document-slack-integ--docs-codacy.netlify.app/organizations/using-coding-standards/)).

## Installing the Slack integration

To install the Slack integration:

> 1.  Follow the [instructions on the Slack documentation pages](https://api.slack.com/messaging/webhooks) to add an Incoming Webhook to your Slack workspace and set it to post to a channel of your choice, and then copy the webhook URL.

Reviewer: Ideally, we should have only one action per step. We have three actions in this step (follow the instructions, set it, copy). We could leave the first together in this step, as they are executed on Slack side, and place the copy in a separate step. WDYT?

```suggestion
1.  Follow the [instructions on the Slack documentation pages](https://api.slack.com/messaging/webhooks) to add an Incoming Webhook to your Slack workspace and set it to post to a channel of your choice.
1.  Copy the webhook URL.
```

Author: Sounds good, I'm extending this with another step.

    The Incoming Webhook URL starts with `https://hooks.slack.com/services/`.

> 1.  Open your organization **Integrations**, page **Slack**, paste the Incoming Webhook URL in the field, and click **Install Slack**.

Reviewer: Same here, we could place the click action on a different step.

```suggestion
1.  Open your organization **Integrations**, page **Slack** and paste the Incoming Webhook URL in the field.
1.  Click **Install Slack**.
```

Author: This one would probably be too granular. We're already using the "and click X" formula in [multiple places](https://github.com/search?q=repo%3Acodacy%2Fdocs%20%22and%20click%22&type=code).

Reviewer: I agree with having the "click X" together with one previous (simple) step.
But in this case, we have two other steps before the click: open and paste. That was the only reason I suggested to separate. 🤓 
Anyway, we can keep it as is as discuss this detail later on. 👍

    ![Slack integration installation](images/slack-integration-install.png)

Once the Slack integration is installed, Codacy sends a confirmation message to the Slack channel you configured when creating the Incoming Webhook.

> ## Uninstalling the Slack integration

Reviewer: Before describing the uninstall use case, I miss the use case showing the value of the integration... 🤔 
After installing the integration, what can the user do/see?

Author: OK, I'm extending the paragraph above to reiterate the main use case, plus adding a screenshot as suggested by @mrfyda 👍

# Organization Slack integration for Security issues

> The organization Slack integration for Security issues sends notifications to a Slack channel of your choice whenever Codacy detects new critical Security issues in any repository within your organization.

Reviewer 2: ```suggestion
The organization Slack integration for Security issues sends notifications to a Slack channel of your choice whenever Codacy detects new critical Security issues in the default branch of any repository in your organization.
```

# Organization Slack integration for Security issues

The organization Slack integration for Security issues sends notifications to a Slack channel of your choice whenever Codacy detects new critical Security issues in the default branch of any repository in your organization.

## Installing the Slack integration

To install the Slack integration:

1.  **On Slack**:

>     1.  [Create an Incoming Webhook](https://api.slack.com/messaging/webhooks) and set it to post to a channel of your choice.

Reviewer 2: It's a good question. I assumed it was different because it's trying to explain the concept of Incoming Webhooks, not starting the flow for integrating. As you said, that flow already happens and is easier to follow from the UI.

```suggestion
    1.  Create an [Incoming Webhook](https://api.slack.com/messaging/webhooks) and set it to post to a channel of your choice.
```

>     1.  [Create an Incoming Webhook](https://api.slack.com/messaging/webhooks) and set it to post to a channel of your choice.

Author: cc @Afonsobli 🙏

>     1.  [Create an Incoming Webhook](https://api.slack.com/messaging/webhooks) and set it to post to a channel of your choice.

Author: [Updated](https://github.com/codacy/docs/blob/839c875079f8eebdb37f578f152f051159b1d872/docs/organizations/integrations/slack-integration.md?plain=1#L10-L14) 👍

    1.  From the page [Your Apps](https://api.slack.com/apps), click the application linked to the webhook you've just created and open the associated **Incoming Webhooks** page.

    1.  Copy the Incoming Webhook URL (it starts with `https://hooks.slack.com/services/`).

> 1.  **On Codacy**:

Reviewer 3: Also, since we provide all these links, we can just point users to the Integration page directly and explain the step by step from there to make it simpler, perhaps

Author: Yes, that's a dynaimc link 😕

[We often use](https://github.com/search?q=repo%3Acodacy%2Fdocs+%22your+organization+**%22&type=code) this pattern in the docs.
